cmd/serve/dlna/dlna.go

```diff
@@ -279,7 +279,14 @@ func (s *server) resourceHandler(w http.ResponseWriter, r *http.Request) {
 // use s.Wait() to block on the listener indefinitely.
 func (s *server) Serve() (err error) {
 	if s.HTTPConn == nil {
-		s.HTTPConn, err = net.Listen("tcp", s.httpListenAddr)
+		// Specifying network as tcp in go will cause -
+		// the bind address 0.0.0.0 to listen to [::] at the same time.
```

Comment:
Binding to 0.0.0.0 is also a global address. Its IPv4 only, whereas [::] is IPv6 only.

Home networks are likely to have a public IPv6 but a private IPv4 so in this case it might improve security, but it doesn't in the general case. Also I'd like to think that even home networks have a firewall for IPv6.

I see your comment later that ssdp only listens on the IPv4 multicast address which I think is a better reason - maybe the comment should be updated to that?

Reply:
```patch
index 604e91f9b..579d695d8 100644
--- a/cmd/serve/dlna/dlna.go
+++ b/cmd/serve/dlna/dlna.go
@@ -279,11 +279,11 @@ func (s *server) resourceHandler(w http.ResponseWriter, r *http.Request) {
 // use s.Wait() to block on the listener indefinitely.
 func (s *server) Serve() (err error) {
        if s.HTTPConn == nil {
-               // Specifying network as tcp in go will cause -
-               // the bind address 0.0.0.0 to listen to [::] at the same time.
-               // And the latter may be a global address, so there is a security risk
+               // Currently, the SSDP server only listens on an IPv4 multicast address.
+               // Differentiate between two INADDR_ANY addresses,
+               // so that 0.0.0.0 can only listen on IPv4 addresses.
                network := "tcp4"
```
Is it ok to comment like this?

Reply:
Looks great